docs/using-react-redux/annotating-connected-components-with-flow.md

```diff
@@ -0,0 +1,177 @@
+---
+id: annotating-connected-components-with-flow
+title: Annotating Connected Components with Flow
+hide_title: true
+sidebar_label: Annotating Connected Components with Flow
+---
+
+# Annotating Connected Components with Flow
+
+After Flow 0.85, Flow starts [Asking for Required Annotations](https://medium.com/flow-type/asking-for-required-annotations-64d4f9c1edf8) on implicit calls of higher order components within each file import — export cycle. This facilitates Flow to merge type information from file dependencies _before_ it walks the type structure and conducts type inference.
+
+This helps Flow gain significantly better coverage on higher order components. But it also asks that we explicitly annotate the connected components at file exports. Exporting implicit calls of `connect` will raise error:
+
+    Missing type annotation for OP. OP is a type parameter declared in function type [1] and was implicitly
+    instantiated at call of connect [2].
+
+In general, to make Flow happy with connect after 0.85 is a two-phase fix. First, you need to explicitly annotate each connected components at file exports. This shall clear all the “implicitly instantiated” errors. Then, if your codebase contains mismatched types between component definitions and usages, Flow will report those errors after you fix the implicit instantiation errors.
+
+## Fixing the “implicitly instantiated” errors at calls of `connect`
+
+To begin, you need:
+
+- Flow upgraded to v0.89+
+- [Newest Flow Typed library definition for React Redux](https://github.com/flow-typed/flow-typed/blob/master/definitions/npm/react-redux_v5.x.x/flow_v0.89.x-/react-redux_v5.x.x.js)
+
+We include three major use cases for annotating `connect` with Flow:
+
+- Connecting stateless component with `mapStateToProps`
+- Connecting components with `mapDispatchToProps` of action creators
+- Connecting components with `mapStateToProps` and `mapDispatchToProps` of action creators
+
+### Connecting stateless component with `mapStateToProps`
+
+```jsx
+type OwnProps = {|
+  passthrough: number,
+  forMapStateToProps: string,
+|};
+type Props = {
+  ...OwnProps,
+  fromStateToProps: string
+};
```

Comment:
Here you mix exact & inexact objects. I'd stick with either exact or inexact (but would definitely prefer exact objects as that's the future proof way)

Reply:
Same with several other places.